File: DAL/DbEntities/Reservation.cs
using Shared.Enums;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace DAL.DbEntities
{
    public class Reservation
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public UserClearance RequiredClearance { get; set; }
        public virtual Room Room { get; set; }
        public int RoomId { get; set; }
        public virtual User Initiator { get; set; }
        public int InitiatorId { get; set; }

        public override string ToString()
        {
            return $"{nameof(Id)}: {Id}, {nameof(Room)}: {RoomId}, {nameof(Initiator)}: {InitiatorId}";
        }
    }
}

File: NgClient/src/app/model/reservation.ts
import { UserClearance } from '../common/enums/user-clearance';
import { ReservationDto } from '../common/dto/reservation.dto';

export class Reservation {
	public constructor(reservationDto: ReservationDto) {
		this.roomDescription = reservationDto.room.description;
		this.startTime = reservationDto.details.startTime;
		this.endTime = reservationDto.details.endTime;
		this.requiredClearance = reservationDto.details.requiredClearance;
	}

	public roomDescription: string;
	public startTime: Date;
	public endTime: Date;
	public requiredClearance: UserClearance;
}